Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -824,7 +824,7 @@ extension PurchasesOrchestrator: StoreKit2TransactionListenerDelegate {
         _ listener: StoreKit2TransactionListener,
         updatedTransaction transaction: StoreTransactionType
     ) async throws {
-        let storefront = await Storefront.currentStorefront
+        let storefront = await transaction.storefront ??? (await Storefront.currentStorefront)
```

Comment:
FFTI I wonder if we should move this inside the `transaction.storefront` calculation, in case we ever need it somewhere else.

Reply:
Yeah let me put that in an extension 👍🏻